### PR TITLE
Add support for IDispatchConstantAttribute

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
cc @jkotas @jeffschwMSFT

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test Linux arm64 Release Build please
@dotnet-bot test OSX x64 Debug Build please

**Comment by jkotas** [comment]
@dotnet-bot test Linux arm64 Release Build please
@dotnet-bot test OSX x64 Debug Build please
@dotnet-bot test Packaging All Configurations x64 Debug Build please

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot tests Linux arm64 Release Build please

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test Linux arm64 Release Build please

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test OSX x64 Debug Build please

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test Linux arm64 Release Build please
@dotnet-bot test OSX x64 Debug Build please
@dotnet-bot test Packaging All Configurations x64 Debug Build please

**Comment by AaronRobinsonMSFT** [comment]
@jkotas @terrajobst I have a question here. Currently `IUnknownConstantAttribute` is in netstandard, but the corresponding `IDispatchConstantAttribute` isn't. I updated the `IDispatchConstantAttribute` test to not build against netstandard, which is correct per existing API, but is this what we want long term? Personally I don't think either should be in netstandard, but I am unclear if we can change that now. I have a few options:

1) Add `IDispatchConstantAttribute` to netstandard
2) Remove `IUnknownConstantAttribute` from netstandard making the pair consistent
3) Leave this as is and move on with life

Thoughts?

cc @jeffschwMSFT 

https://github.com/dotnet/standard/blob/e9f1d2d3d08b7f19ba6f4704fb4e2401fe504bf8/netstandard/ref/System.Runtime.CompilerServices.cs#L396-L400

**Comment by jkotas** [comment]
Leave this as is and move on with life

Removing from netstandard is not an option. Potential additions to netstandard are done on separate schedule and typically in bulk.

File: src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj [hunk 1]
```diff
@@ -131,6 +131,7 @@
     <Compile Include="System\Runtime\InteropServices\VariantWrapperTests.cs" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetGroup)' != 'netstandard'">
+    <Compile Include="System\Runtime\CompilerServices\IDispatchConstantAttributeTests.cs"  Condition="'$(TargetsWindows)' == 'true'" />
```
Comment (jkotas) [inline src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj]: The preferred CoreFX style is to exclude the platform-specific tests with attributes, like `[PlatformSpecific(TestPlatforms.Windows)]`.

Comment (AaronRobinsonMSFT) [reply src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj]: Will do. Thanks.

**Pull request merged**